Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Plan.hs

```diff
@@ -0,0 +1,303 @@
+  mandate <- validateInActiveMandateExists driverId driverPlan
+  Esq.runTransaction $ do
+    QM.updateStatus mandate.id DM.ACTIVE
+    CDI.updateAutoPayStatus (Just DI.ACTIVE) (cast driverId)
+  return Success
+
+---------------------------------------------------------------------------------------------------------
+------------------------------------------ Helper Functions ---------------------------------------------
+---------------------------------------------------------------------------------------------------------
+
+validateActiveMandateExists :: Id SP.Person -> DriverPlan -> Flow DM.Mandate
+validateActiveMandateExists driverId driverPlan = do
+  case driverPlan.mandateId of
+    Nothing -> throwError $ ActiveMandateDoNotExist driverId.getId
+    Just mandateId -> do
+      mandate <- Esq.runInReplica $ QM.findById mandateId >>= fromMaybeM (MandateNotFound mandateId.getId)
+      unless (mandate.status == DM.INACTIVE) $ throwError (ActiveMandateDoNotExist driverId.getId)
+      return mandate
+
+validateInActiveMandateExists :: Id SP.Person -> DriverPlan -> Flow DM.Mandate
+validateInActiveMandateExists driverId driverPlan = do
+  case driverPlan.mandateId of
+    Nothing -> throwError $ ActiveMandateDoNotExist driverId.getId
+    Just mandateId -> do
+      mandate <- Esq.runInReplica $ QM.findById mandateId >>= fromMaybeM (MandateNotFound mandateId.getId)
+      unless (mandate.status == DM.ACTIVE) $ throwError (ActiveMandateDoNotExist driverId.getId)
+      return mandate
+
+validateActiveMandateDoNotExists :: Id SP.Person -> DriverPlan -> Flow ()
+validateActiveMandateDoNotExists driverId driverPlan = do
+  case driverPlan.mandateId of
+    Just mandateId -> do
+      mandate <- Esq.runInReplica $ QM.findById mandateId >>= fromMaybeM (MandateNotFound mandateId.getId)
+      unless (mandate.status == DM.ACTIVE) $ throwError (ActiveMandateExists driverId.getId)
+    Nothing -> pure ()
+
+createMandateInvoiceAndOrder :: Id SP.Person -> Id DM.Merchant -> Plan -> Flow (Payment.CreateOrderResp, Id DOrder.PaymentOrder)
+createMandateInvoiceAndOrder driverId merchantId plan = do
+  driverFees <- QDF.findAllPendingAndDueDriverFeeByDriverId driverId
+  if not (null driverFees)
+    then SPayment.createOrder (driverId, merchantId) driverFees (Just mandateOrder)
+    else do
+      driverFee <- mkDriverFee
+      Esq.runNoTransaction $ QDF.create driverFee
+      SPayment.createOrder (driverId, merchantId) [driverFee] (Just mandateOrder)
+  where
+    mandateOrder =
+      SPayment.MandateOrder
+        { maxAmount = plan.maxAmount,
+          _type = Payment.REQUIRED,
+          frequency = Payment.ASPRESENTED
+        }
+    mkDriverFee = do
+      id <- generateGUID
+      now <- getCurrentTime
+      return $
+        DF.DriverFee
+          { id = id,
+            merchantId = merchantId,
+            payBy = now,
```

Comment:
The rule is to make driver unsubscribed if they don't pay after `payBy`. Are we not following that here?

Reply:
No @0utkarsh For now while rolling out this feature we will not generate invoices for drivers. That will start after september 1st.